bigquery/google/cloud/bigquery/client.py

```diff
@@ -186,51 +186,53 @@ def list_projects(self, max_results=None, page_token=None,
             page_token=page_token,
             max_results=max_results)
 
-    def list_datasets(self, include_all=False, filter=None, max_results=None,
-                      page_token=None, retry=DEFAULT_RETRY):
+    def list_datasets(
+            self, project=None, include_all=False, filter=None,
+            max_results=None, page_token=None, retry=DEFAULT_RETRY):
```

Comment:
FWIW, putting the new argument last in the list provides somewhat better backward-compatibility for users who may not have used named parameters.  I'm not sure how important that is in this case.

Reply:
Yeah, I wasn't sure. Looks like we want `retry` to be the last arg in `Client` methods. Since it's the resource you are listing over, the first place felt the most natural.

Per the last time I tried to preserve positional/kwarg semantics, it sounds like we consider these args just keyword arguments even though they could use either.

Reply:
I'm fine to merge if you think the switch to `1.0` doesn't tighten up that constraint.